src/main/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFBlockMergingIterator.java

```diff
@@ -0,0 +1,17 @@
+package org.broadinstitute.hellbender.utils.variant.writers;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.utils.iterators.PushToPullIterator;
+
+import java.util.*;
+
+/**
+ * Turns an iterator of {@link VariantContext} into one which merges GVCF blocks.
+ */
+public class GVCFBlockMergingIterator extends PushToPullIterator<VariantContext> {
```

Comment:
This is unreferenced; its unclear to me if thats deliberate because its intended to be used independently of GVCFWriter ? Also, I would suggest unifying the names of `GVCFBlockMergingIterator ` and `GVCFBlockCombiner ` so they both use the same verb, either `combine` or `merge`.

Reply:
It's intended to be used separately, and in fact the Disq branch that follows this one uses it (#5138). Also renamed `GVCFBlockMergingIterator` to `GVCFBlockCombiningIterator`.